File: src/main/kotlin/Java/Estudiantes/Instituto.java
package Java.Estudiantes;

import java.util.ArrayList;
import java.util.List;

public class Instituto {
    private List<Estudiante> estudiantes;
    private List<Profesor> profesores;

    public Instituto() {
        estudiantes = new ArrayList<>();
        profesores = new ArrayList<>();
    }

    public void registrarEstudiante(Estudiante estudiante) {
        estudiantes.add(estudiante);
        System.out.println("Estudiante registrado: " + estudiante.getNombre());
    }

    public void registrarProfesor(Profesor profesor) {
        profesores.add(profesor);
        System.out.println("Profesor registrado: " + profesor.getNombre());
    }

    public void mostrarCalificaciones() {
        estudiantes.forEach(estudiante -> {
            System.out.println("Estudiante: " + estudiante.getNombre() + ", Promedio: " + estudiante.calificacionPromedio());
        });
    }

    // Getters y Setters
    public List<Estudiante> getEstudiantes() {
        return estudiantes;
    }

    public void setEstudiantes(List<Estudiante> estudiantes) {
        this.estudiantes = estudiantes;
    }

    public List<Profesor> getProfesores() {
        return profesores;
    }

    public void setProfesores(List<Profesor> profesores) {
        this.profesores = profesores;
    }
}

File: src/main/kotlin/Estudiantes/Instituto.kt
package Estudiantes

class Instituto {
    val estudiantes = mutableListOf<Estudiante>()
    val profesores = mutableListOf<Profesor>()

    fun registrarEstudiante(estudiante: Estudiante) {
        estudiantes.add(estudiante)
        println("Estudiante registrado: ${estudiante.nombre}")
    }

    fun registrarProfesor(profesor: Profesor) {
        profesores.add(profesor)
        println("Profesor registrado: ${profesor.nombre}")
    }

    fun mostrarCalificaciones() {
        estudiantes.forEach { estudiante ->
            println("Estudiante: ${estudiante.nombre}, Promedio: ${estudiante.calificacionPromedio()}")
        }
    }
}
